uat/test_uat_CLIParser.py

```diff
@@ -0,0 +1,53 @@
+import os
+import subprocess as sp
+import tempfile
+from pathlib import Path
+
+import gdk.common.exceptions.error_messages as error_messages
+
+
+def install_cli():
+    sp.run(["pip3", "install", "git+https://github.com/aws-greengrass/aws-greengrass-gdk-cli.git"])
```

Comment:
this will install from HEAD of the main branch. I think we may want more smartness around this.

For example, installing from the local version ie. when running a PR.